components/swim/src/server/inbound.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::mpsc;
+use std::sync::atomic::Ordering;
+use std::net::SocketAddr;
+use std::thread;
+use std::time::Duration;
+
+use protobuf;
+
+use message::swim::{Swim, Swim_Type};
+use server::{Server, outbound};
+use member::{Member, Health};
+
+pub struct Inbound<'a> {
+    pub server: &'a Server,
+    pub tx_outbound: mpsc::Sender<(SocketAddr, Swim)>,
+}
+
+impl<'a> Inbound<'a> {
+    pub fn new(server: &'a Server, tx_outbound: mpsc::Sender<(SocketAddr, Swim)>) -> Inbound {
+        Inbound {
+            server: server,
+            tx_outbound: tx_outbound
+        }
+    }
+
+    pub fn run(&self) {
+        let mut recv_buffer: Vec<u8> = vec![0; 1024];
+        loop {
+            if self.server.pause.load(Ordering::Relaxed) {
+                thread::sleep(Duration::from_millis(100));
+                continue;
+            }
+            match self.server.socket.recv_from(&mut recv_buffer[..]) {
+                Ok((length, addr)) => {
+                    if self.server.check_blacklist(&addr) {
+                        println!("Not processing message from {} - it is blacklisted", addr);
+                        continue;
+                    }
+                    let msg: Swim = match protobuf::parse_from_bytes(&recv_buffer[0..length]) {
+                        Ok(msg) => msg,
+                        Err(e) => {
+                            error!("Error parsing protobuf: {:?}", e);
```

Comment:
That could be a thing we use the blacklist for eventually. I'm inclined to take this as a future note, and see if we want to add it later.